### ISSUE TITLE
What should I name the Commit class of a PushEventPayload?

### ISSUE THREAD

**Issue by itaibh** [octokit/octokit.net#1789]
The PushEvent payload have a different `Commit` class structure than just about anywhere else (the JSON samples are from the official docs):

**PushEvent Payload:**
```json
"commits": [
    {
      "id": "0d1a26e67d8f5eaf1f6ba5c57fc3c7d91ac0fd1c",
      "tree_id": "f9d2a07e9488b91af2641b26b9407fe22a451433",
      "distinct": true,
      "message": "Update README.md",
      "timestamp": "2015-05-05T19:40:15-04:00",
      "url": "https://github.com/baxterthehacker/public-repo/commit/0d1a26e67d8f5eaf1f6ba5c57fc3c7d91ac0fd1c",
      "author": {
        "name": "baxterthehacker",
        "email": "baxterthehacker@users.noreply.github.com",
        "username": "baxterthehacker"
      },
      "committer": {
        "name": "baxterthehacker",
        "email": "baxterthehacker@users.noreply.github.com",
        "username": "baxterthehacker"
      },
      "added": [

      ],
      "removed": [

      ],
      "modified": [
        "README.md"
      ]
    }
  ],
  "head_commit": {
    "id": "0d1a26e67d8f5eaf1f6ba5c57fc3c7d91ac0fd1c",
    "tree_id": "f9d2a07e9488b91af2641b26b9407fe22a451433",
    "distinct": true,
    "message": "Update README.md",
    "timestamp": "2015-05-05T19:40:15-04:00",
    "url": "https://github.com/baxterthehacker/public-repo/commit/0d1a26e67d8f5eaf1f6ba5c57fc3c7d91ac0fd1c",
    "author": {
      "name": "baxterthehacker",
      "email": "baxterthehacker@users.noreply.github.com",
      "username": "baxterthehacker"
    },
    "committer": {
      "name": "baxterthehacker",
      "email": "baxterthehacker@users.noreply.github.com",
      "username": "baxterthehacker"
    },
    "added": [

    ],
    "removed": [

    ],
    "modified": [
      "README.md"
    ]
  },
```

**Just about anywhere else:**
```json
 {
    "sha": "47045ee9724ab11d0249bae75a599dd3ebd10a09",
    "commit": {
      "author": {
        "name": "Itai Bar-Haim",
        "email": "itaibh@gmail.com",
        "date": "2018-01-01T09:14:48Z"
      },
      "committer": {
        "name": "GitHub",
        "email": "noreply@github.com",
        "date": "2018-01-01T09:14:48Z"
      },
      "message": "Merge pull request #1 from itaibh/itaibh-patch-1\n\nUpdate README.md",
      "tree": {
        "sha": "d23387f363f3781160241279b50d03a08a6624ac",
        "url": "https://api.github.com/repos/itaibh/Spoon-Knife/git/trees/d23387f363f3781160241279b50d03a08a6624ac"
      },
      "url": "https://api.github.com/repos/itaibh/Spoon-Knife/git/commits/47045ee9724ab11d0249bae75a599dd3ebd10a09",
      "comment_count": 0,
      "verification": {
        "verified": true,
        "reason": "valid",
        "signature": "-----BEGIN PGP SIGNATURE-----\n\nwsBcBAABCAAQBQJaSfwICRBK7hj4Ov3rIwAAdHIIAIi/QCdPQ/nHCLApqXbmbbUi\nMrpd4GSHTH1BnpIZBZVfFV38/wbsEGJk+UG0YGSiX6pEWwOTapCFlCyFd5MNDtwu\nHILfCa1jgv28bwh6Nl18b7Gqx2fKq5JJeekHIO4FJyShhfTQD/nCo202cJZdwgHc\nefVwwDvJ8kTPzb1KTSDAeoKQVVHqXPSTWzTQGYF5PuqktjLhSg3fIx8UhnT+dJQF\nlqxy2GXFqax0gndC+LnjV+d13MymYY9/G82ji/CLFRzWHNbyICsnZC/7BIHzi5Iu\nKGFP0tFE3d35RFqhc5/0qw/npDjfvt0xD25ixj5i/7cumgY+LH8bU8Y0E3P2qtg=\n=GPAE\n-----END PGP SIGNATURE-----\n",
        "payload": "tree d23387f363f3781160241279b50d03a08a6624ac\nparent 8340ef42a8fc9915563396820b7cedc7424e3a84\nparent 1659dd0636941523600be861fd6015f7d48a4e76\nauthor Itai Bar-Haim <itaibh@gmail.com> 1514798088 +0200\ncommitter GitHub <noreply@github.com> 1514798088 +0200\n\nMerge pull request #1 from itaibh/itaibh-patch-1\n\nUpdate README.md"
      }
    },
    "url": "https://api.github.com/repos/itaibh/Spoon-Knife/commits/47045ee9724ab11d0249bae75a599dd3ebd10a09",
    "html_url": "https://github.com/itaibh/Spoon-Knife/commit/47045ee9724ab11d0249bae75a599dd3ebd10a09",
    "comments_url": "https://api.github.com/repos/itaibh/Spoon-Knife/commits/47045ee9724ab11d0249bae75a599dd3ebd10a09/comments",
    "author": {
      "login": "itaibh",
      "id": 140983,
      "avatar_url": "https://avatars3.githubusercontent.com/u/140983?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/itaibh",
      "html_url": "https://github.com/itaibh",
      "followers_url": "https://api.github.com/users/itaibh/followers",
      "following_url": "https://api.github.com/users/itaibh/following{/other_user}",
      "gists_url": "https://api.github.com/users/itaibh/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/itaibh/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/itaibh/subscriptions",
      "organizations_url": "https://api.github.com/users/itaibh/orgs",
      "repos_url": "https://api.github.com/users/itaibh/repos",
      "events_url": "https://api.github.com/users/itaibh/events{/privacy}",
      "received_events_url": "https://api.github.com/users/itaibh/received_events",
      "type": "User",
      "site_admin": false
    },
    "committer": {
      "login": "web-flow",
      "id": 19864447,
      "avatar_url": "https://avatars3.githubusercontent.com/u/19864447?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/web-flow",
      "html_url": "https://github.com/web-flow",
      "followers_url": "https://api.github.com/users/web-flow/followers",
      "following_url": "https://api.github.com/users/web-flow/following{/other_user}",
      "gists_url": "https://api.github.com/users/web-flow/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/web-flow/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/web-flow/subscriptions",
      "organizations_url": "https://api.github.com/users/web-flow/orgs",
      "repos_url": "https://api.github.com/users/web-flow/repos",
      "events_url": "https://api.github.com/users/web-flow/events{/privacy}",
      "received_events_url": "https://api.github.com/users/web-flow/received_events",
      "type": "User",
      "site_admin": false
    },
    "parents": [
      {
        "sha": "8340ef42a8fc9915563396820b7cedc7424e3a84",
        "url": "https://api.github.com/repos/itaibh/Spoon-Knife/commits/8340ef42a8fc9915563396820b7cedc7424e3a84",
        "html_url": "https://github.com/itaibh/Spoon-Knife/commit/8340ef42a8fc9915563396820b7cedc7424e3a84"
      },
      {
        "sha": "1659dd0636941523600be861fd6015f7d48a4e76",
        "url": "https://api.github.com/repos/itaibh/Spoon-Knife/commits/1659dd0636941523600be861fd6015f7d48a4e76",
        "html_url": "https://github.com/itaibh/Spoon-Knife/commit/1659dd0636941523600be861fd6015f7d48a4e76"
      }
    ]
  },
```

As you can see, the PushEvent payload's commit structure contains `id` instead of `sha`, and have `tree_id`, `added`, `removed` and `modified` fields.

According to here https://platform.github.community/t/payload-documentation-doesnt-match-actual-payloads/5388/2 , it seems that GitHub isn't going to fix that anytime soon, so I must create a new class, and I figured I should consult the name of it.

For the meanwhile I'll just use `Commit2` (I think I saw this structure in another payload too, so `PushCommit` is not a good name), but I'll change it once we agree on a new name.

**Comment by ryangribble** [comment]
Maybe `CommitPayload` if it appears in places other than push event

**Comment by itaibh** [comment]
Fixed in #1844
